conans/model/conan_file.py

```diff
@@ -77,8 +79,15 @@ def create_exports_sources(conanfile):
         return conanfile.exports_sources
 
 
+@contextmanager
+def _env_and_python(conanfile):
+    with environment_append(conanfile.env):
+        with pythonpath(conanfile):
```

Comment:
This might be too slow to be done in all places, by default.

Reply:
Did you manage to do some performance test?

Reply:
3,5% of extra time.

The context is being applied to ``__config_node()`` in ``DepsBuilder`` which does not apply there, maybe that could be omitted.

Reply:
Could you repeat the measure now? I excluded the python env from the deps builder

Reply:
(pending tests)